File: blog/src/main/java/com/github/blog/controller/util/marker/UserValidationGroup.java
package com.github.blog.controller.util.marker;

import jakarta.validation.GroupSequence;
import jakarta.validation.groups.Default;

/**
 * @author Raman Haurylau
 */
public interface UserValidationGroup {

    @GroupSequence({BaseMarker.First.class, Default.class, BaseMarker.Second.class, BaseMarker.Third.class})
    interface onCreate {
    }

    @GroupSequence({BaseMarker.First.class, Default.class,})
    interface onAuthenticate {
    }

    @GroupSequence({Default.class, BaseMarker.Second.class, BaseMarker.Third.class})
    interface onUpdate {
    }
}

File: api-gateway/src/main/kotlin/com/github/api_gateway/controller/util/marker/UserValidationGroup.kt
package com.github.api_gateway.controller.util.marker

import jakarta.validation.GroupSequence
import jakarta.validation.groups.Default

/**
 * @author Raman Haurylau
 */
interface UserValidationGroup {
    @GroupSequence(value = [BaseMarker.First::class, Default::class, BaseMarker.Second::class, BaseMarker.Third::class])
    interface onCreate

    @GroupSequence(value = [Default::class, BaseMarker.Second::class, BaseMarker.Third::class])
    interface onUpdate

    @GroupSequence(value = [BaseMarker.First::class, Default::class])
    interface onAuthenticate
}